### PR TITLE
transport: notify snapshot failure on resolving

### PR DESCRIPTION
**Pull request by overvenus**
When resolving remote address, all messages will be dropped and
report unreachable. However unreachable won't reset follower's
progress if it's in Snapshot state. So trying to send a snapshot
when the address is being resolved will leave follower's progress
stay in Snapshot forever.

### CODE REVIEW

**Comment by overvenus** [comment]
A friendly ping.

File: tests/failpoints_cases/test_snap.rs [hunk 1]
```diff
@@ -66,3 +74,106 @@ fn test_overlap_cleanup() {
...
+            debug!("notify snapshot");
+            self.pending_notify.fetch_sub(1, Ordering::SeqCst);
+            self.notifier.lock().unwrap().send(()).unwrap();
+        }
+        Ok(())
+    }
+}
+
+#[test]
+fn test_server_snapshot_on_reslove_failure() {
```
Comment (BusyJay) [inline tests/failpoints_cases/test_snap.rs]: Please comment what the case we have encountered.

File: tests/failpoints_cases/test_snap.rs [hunk 2]
```diff
@@ -66,3 +74,111 @@ fn test_overlap_cleanup() {
...
+                && self.ready_notify.load(Ordering::SeqCst)
+            {
+                self.pending_notify.fetch_add(1, Ordering::SeqCst);
+            }
+        }
+
+        Ok(())
+    }
+
+    fn after(&self, _: Result<()>) -> Result<()> {
```
Comment (huachaohuang) [inline tests/failpoints_cases/test_snap.rs]: Should we check message type here?

Comment (overvenus) [reply tests/failpoints_cases/test_snap.rs]: No, we have checked in the `before` and we want to send notifications after sending messages.

**Comment by huachaohuang** [comment]
LGTM

**Pull request merged**